src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -41,20 +40,36 @@ export class RundownComponent implements OnInit, OnDestroy {
   }
 
   private onRundownTimingContextChanged(rundownTimingContext: RundownTimingContext): void {
-    this.onAirSegmentIdForRundownTimingContext = rundownTimingContext.onAirSegmentId
+    this.currentEpochTime = rundownTimingContext.currentEpochTime
 
     const onAirPart: Part | undefined = this.rundown.segments.find(segment => segment.isOnAir)?.parts.find(part => part.isOnAir)
 
-    if (!onAirPart || !this.doesPartContainVideoClipOrVoiceOver(onAirPart)) {
+    if (!onAirPart) {
       this.remainingDurationInMsForOnAirPart = undefined
     } else {
-      this.remainingDurationInMsForOnAirPart = rundownTimingContext.playedDurationInMsForOnAirPart - this.partEntityService.getExpectedDuration(onAirPart)
+      this.remainingDurationInMsForOnAirPart = this.partEntityService.getExpectedDuration(onAirPart) - rundownTimingContext.playedDurationInMsForOnAirPart
+    }
```

Comment:
Could be a ternary.

Also, you make the exact same check twice right after each other.
Since you are not using ternarys you should combine the two if-statements.

Reply:
Is changed to a ternary.